packages/tpetra/core/src/Tpetra_Details_packCrsMatrix.hpp

```diff
@@ -70,17 +72,181 @@ class Distributor;
 //
 namespace Details {
 
-template<class LocalSparseMatrixType>
+
+/// \brief Compute number of packets per row of the local matrix
+///
+/// \tparam GO Global ordinal type
+/// \tparam ViewType Type of the Kokkos::View specialization
+///   used to store the number of packets; the output array of this function.
+/// \tparam ExportLIDsViewType Type of the Kokkos::View specialization
+///   used to store the LIDs of local rows; the input array of this function.
+/// \tparam LocalMatrixType Type of the KokkosSparse::CrsMatrix local matrix
+///
+template<class GO, class ViewType, class ExportLIDsViewType, class LocalMatrixType>
+struct NumPacketsFunctor {
+
+  typedef typename ViewType::value_type value_type;
+  typedef typename LocalMatrixType::row_map_type row_map_type;
+  typedef typename LocalMatrixType::value_type IST;
+  typedef typename LocalMatrixType::ordinal_type LO;
+
+  ViewType num_packets;
+  ExportLIDsViewType export_lids;
+  row_map_type row_map;
+
+  NumPacketsFunctor(ViewType num_packets_, ExportLIDsViewType export_lids_,
```

Comment:
Right.  It seemed a convention (like in the [Kokkos sparse CrsMatrix](https://github.com/kokkos/kokkos-kernels/blob/master/src/Kokkos_Sparse_CrsMatrix.hpp#L124)) had trailing underscores on input arguments that had the same name as (public) member data, to avoid shadowing names.  I hadn't explicitly made the member data of the struct private, so I adopted that convention.  I can swap it if it fits better with Tpetra